## 58% of end-end tests have passed
```
Total executed tests: 52
Failed tests: 21
Expected to fail tests: 1
Passed tests: 30
```
```
IDs of failed tests: 702840,702894,703086,702775,702731,702844,702851,704613,702841,703496,702839,702733,704615,702859,702869,702784,703382,703391,703297,703495,703194 
```
```
IDs of expected to fail tests: 703503 
```
<h3>Failed tests (21)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&PR_ID=19270&TR_CASE_IDS=702840,702894,703086,702775,702731,702844,702851,704613,702841,703496,702839,702733,704615,702859,702869,702784,703382,703391,703297,703495,703194">Rerun failed tests</a></li><br/><h4>Class TestDeepLinksOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518146">test_links_deep_links</a>, id: 702775 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `BrowserTab` by `accessibility id`: `browser-stack-tab` </div><div>Device 1: Tap on found: BrowserTab </div></blockquote></p>

```
critical/test_deep_and_universal_links.py:112: in test_links_deep_links
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Community 'Closed communtiy for e2e' was not requested to join by the deep link status.app://c/G0UAAGS9MeHbo_RoNMnOhOKTCo-fDDbgBCCyNNSN7qrzTqIgLem56TyzYYLf36askV7yGcros8muAdgB#zQ3shVWCcsH94vjf7WrFqBXx5MMAui2EPYMBme8ntACNWWVN3
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/95daf83828f44fe2b66a2e144ac29f5b?auth=c6a921069cb5db0ed3ad5345d83276fb#192">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518186">test_links_open_universal_links_from_chat</a>, id: 704613 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `xpath`: `//*[@text="open community"]` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/test_deep_and_universal_links.py:69: in test_links_open_universal_links_from_chat
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Community 'Closed communtiy for e2e' was not requested to join by the url https://status.app/c/G0UAAGS9MeHbo_RoNMnOhOKTCo-fDDbgBCCyNNSN7qrzTqIgLem56TyzYYLf36askV7yGcros8muAdgB#zQ3shVWCcsH94vjf7WrFqBXx5MMAui2EPYMBme8ntACNWWVN3
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/95daf83828f44fe2b66a2e144ac29f5b?auth=c6a921069cb5db0ed3ad5345d83276fb#53">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518141">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Text` by `xpath`: `//*[starts-with(@text,'Message 1')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc='message-status']/android.widget.TextView` </div><div>Device 1: `Text` is `Sent` </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:187: in test_1_1_chat_pin_messages
    self.chat_1.chat_element_by_text(self.message_1).wait_for_status_to_be("Delivered")
../views/chat_view.py:225: in wait_for_status_to_be
    raise TimeoutException("Message status was not changed to %s, it's %s" % (expected_status, current_status))
 Message status was not changed to Delivered, it's Sent
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cd6dfd37d3a549b3af0bd20e50d215b7?auth=9ee5b081b5ab9ff62bef8044eed85ab4#202">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cbf47048761948e1ad55ef35923db450?auth=ee26ba6a98cbdc7f92f7adbb05f407e4#165">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518143">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><p><blockquote><div>Device 2: Looking for a message by text: smth I should edit </div><div>Device 2: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'smth I should edit')]` </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:497: in test_1_1_chat_text_message_delete_push_disappear
    self.chat_2.delete_message_in_chat(message_after_edit_1_1)
../views/chat_view.py:1046: in delete_message_in_chat
    self.chat_element_by_text(message).message_body.long_press_element()
../views/base_element.py:321: in long_press_element
    element = self.find_element()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 2: Text by xpath: `//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'smth I should edit')]` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cd6dfd37d3a549b3af0bd20e50d215b7?auth=9ee5b081b5ab9ff62bef8044eed85ab4#449">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cbf47048761948e1ad55ef35923db450?auth=ee26ba6a98cbdc7f92f7adbb05f407e4#367">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518181">test_1_1_chat_send_image_save_and_share</a>, id: 703391 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Button` by `accessibility id`: `image-0` </div><div>Device 2: Click system back button </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:458: in test_1_1_chat_send_image_save_and_share
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Not expected image is shown to the receiver.
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cd6dfd37d3a549b3af0bd20e50d215b7?auth=9ee5b081b5ab9ff62bef8044eed85ab4#388">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cbf47048761948e1ad55ef35923db450?auth=ee26ba6a98cbdc7f92f7adbb05f407e4#307">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518179">test_group_chat_send_image_save_and_share</a>, id: 703297 </b></td></tr><tr><td><p><blockquote><div>Device 3: Find `ChatsTab` by `accessibility id`: `chats-stack-tab` </div><div>Device 3: Tap on found: ChatsTab </div></blockquote></p>

```
critical/chats/test_group_chat.py:283: in test_group_chat_send_image_save_and_share
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Not expected image is shown to the admin.
E    Not expected image is shown to the member_2.
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/217ed82bf5ec4501b656d811de58b2ce?auth=f84a7c422bb33fd6cc3d96a050fcce1c#265">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/26a03bcb395648359c255998c411ac29?auth=35e1484739041a09955f357f308feb19#175">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5b2f2e65c12641e09c8869165ca788e8?auth=cbb8ea813aab7b6fe165af7de4a18c52#146">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518182">test_group_chat_mute_chat</a>, id: 703495 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: Button </div><div>Device 2: Looking for chat: 'gscsyiw' </div></blockquote></p>

```
critical/chats/test_group_chat.py:419: in test_group_chat_mute_chat
    chat.long_press_element()
../views/base_element.py:321: in long_press_element
    element = self.find_element()
../views/home_view.py:74: in find_element
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: ChatElement by xpath:`//*[@content-desc='author-primary-name'][starts-with(@text,'gscsyiw')]/..` is not found on the screen after wait_for_visibility_of_element; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/217ed82bf5ec4501b656d811de58b2ce?auth=f84a7c422bb33fd6cc3d96a050fcce1c#470">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/26a03bcb395648359c255998c411ac29?auth=35e1484739041a09955f357f308feb19#286">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5b2f2e65c12641e09c8869165ca788e8?auth=cbb8ea813aab7b6fe165af7de4a18c52#257">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518158">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><p><blockquote><div>Device 2: Quoting '🎗' message </div><div>Device 2: Find `BaseElement` by `xpath`: `//*[@content-desc=':chat-floating-screen']//*[starts-with(@text,'🎗')]` </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:522: in test_community_emoji_send_copy_paste_reply
    self.channel_2.quote_message(emoji_unicode)
../views/chat_view.py:1057: in quote_message
    self.chat_view_element_starts_with_text(message).long_press_until_element_is_shown(self.reply_message_button)
../views/base_element.py:327: in long_press_until_element_is_shown
    element = self.find_element()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 2: BaseElement by xpath: `//*[@content-desc=':chat-floating-screen']//*[starts-with(@text,'🎗')]` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6bc6d12842314a5d8ffa3e5a030de2df?auth=a702dff2b5678f835e018c1c99fdc9e7#246">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6a0931ee38bc4f2da6681990a55ddfeb?auth=b9d4a42f4273e72d843f5fb8ffdf55de#291">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518169">test_community_contact_block_unblock_offline</a>, id: 702894 </b></td></tr><tr><td><p><blockquote><div>Device 2: Could not reach home view by pressing system back button </div><div>Device 2: Find `CommunitiesTab` by `accessibility id`: `communities-stack-tab` </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:643: in test_community_contact_block_unblock_offline
    self.homes[i].communities_tab.click()
../views/base_element.py:90: in click
    element = self.find_element()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 2: CommunitiesTab by accessibility id: `communities-stack-tab` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6bc6d12842314a5d8ffa3e5a030de2df?auth=a702dff2b5678f835e018c1c99fdc9e7#292">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6a0931ee38bc4f2da6681990a55ddfeb?auth=b9d4a42f4273e72d843f5fb8ffdf55de#417">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518175">test_community_mark_all_messages_as_read</a>, id: 703086 </b></td></tr><tr><td><p><blockquote><div>Device 2: Could not reach home view by pressing system back button </div><div>Device 2: Find `CommunitiesTab` by `accessibility id`: `communities-stack-tab` </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:747: in test_community_mark_all_messages_as_read
    home.communities_tab.click()
../views/base_element.py:90: in click
    element = self.find_element()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 2: CommunitiesTab by accessibility id: `communities-stack-tab` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6bc6d12842314a5d8ffa3e5a030de2df?auth=a702dff2b5678f835e018c1c99fdc9e7#313">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6a0931ee38bc4f2da6681990a55ddfeb?auth=b9d4a42f4273e72d843f5fb8ffdf55de#439">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518161">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td><p><blockquote><div># STEP: Checking github_pr preview case</div><div>Device 2: Find `ChatMessageInput` by `accessibility id`: `chat-message-input` </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:579: in test_community_links_with_previews_github_youtube_twitter_gif_send_enable
    self.channel_2.chat_message_input.send_keys(url)
../views/base_element.py:377: in send_keys
    self.find_element().send_keys(value)
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 2: ChatMessageInput by accessibility id: `chat-message-input` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception; 
 RemoteDisconnected; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6bc6d12842314a5d8ffa3e5a030de2df?auth=a702dff2b5678f835e018c1c99fdc9e7#267">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6a0931ee38bc4f2da6681990a55ddfeb?auth=b9d4a42f4273e72d843f5fb8ffdf55de#302">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518159">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><p><blockquote><div>Device 2: Could not reach home view by pressing system back button </div><div>Device 2: Find `CommunitiesTab` by `accessibility id`: `communities-stack-tab` </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:621: in test_community_unread_messages_badge
    self.home_2.communities_tab.click()
../views/base_element.py:90: in click
    element = self.find_element()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 2: CommunitiesTab by accessibility id: `communities-stack-tab` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6bc6d12842314a5d8ffa3e5a030de2df?auth=a702dff2b5678f835e018c1c99fdc9e7#271">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6a0931ee38bc4f2da6681990a55ddfeb?auth=b9d4a42f4273e72d843f5fb8ffdf55de#395">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518157">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><p><blockquote><div># STEP: Deleting message for me. Checking that message is deleted only for the author of the message</div><div>Device 2: Sending message 'delete for me' </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:375: in test_community_message_delete
    self.channel_2.send_message(message_to_delete_for_me)
../views/chat_view.py:999: in send_message
    self.chat_message_input.wait_for_element(wait_chat_input_sec)
../views/base_element.py:129: in wait_for_element
    raise TimeoutException(
 Device `2`: `ChatMessageInput` by` accessibility id`: `chat-message-input` is not found on the screen after wait_for_element; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6bc6d12842314a5d8ffa3e5a030de2df?auth=a702dff2b5678f835e018c1c99fdc9e7#196">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6a0931ee38bc4f2da6681990a55ddfeb?auth=b9d4a42f4273e72d843f5fb8ffdf55de#136">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518187">test_community_edit_delete_message_when_offline</a>, id: 704615 </b></td></tr><tr><td><p><blockquote><div># STEP: Sending messages for edit and delete</div><div>Device 2: Sending message 'message to edit' </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:776: in test_community_edit_delete_message_when_offline
    self.channel_2.send_message(message_to_edit)
../views/chat_view.py:999: in send_message
    self.chat_message_input.wait_for_element(wait_chat_input_sec)
../views/base_element.py:129: in wait_for_element
    raise TimeoutException(
 Device `2`: `ChatMessageInput` by` accessibility id`: `chat-message-input` is not found on the screen after wait_for_element; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6bc6d12842314a5d8ffa3e5a030de2df?auth=a702dff2b5678f835e018c1c99fdc9e7#334">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6a0931ee38bc4f2da6681990a55ddfeb?auth=b9d4a42f4273e72d843f5fb8ffdf55de#460">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518167">test_community_one_image_send_reply</a>, id: 702859 </b></td></tr><tr><td><p><blockquote><div>Device 2: Looking for a message by text: description </div><div>Device 2: Looking for a message by text: description </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:439: in test_community_one_image_send_reply
    self.channel_2.chat_element_by_text(image_description).wait_for_visibility_of_element(10)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: ChatElementByText by xpath:`//*[starts-with(@text,'description')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']` is not found on the screen after wait_for_visibility_of_element; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6bc6d12842314a5d8ffa3e5a030de2df?auth=a702dff2b5678f835e018c1c99fdc9e7#230">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6a0931ee38bc4f2da6681990a55ddfeb?auth=b9d4a42f4273e72d843f5fb8ffdf55de#272">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518177">test_community_several_images_send_reply</a>, id: 703194 </b></td></tr><tr><td><p><blockquote><div>Device 2: ChatMessageInput element not found </div><div>Device 2: Looking for a message by text: gallery </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:428: in test_community_several_images_send_reply
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Gallery message was not received; 
 RemoteDisconnected; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6bc6d12842314a5d8ffa3e5a030de2df?auth=a702dff2b5678f835e018c1c99fdc9e7#210">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6a0931ee38bc4f2da6681990a55ddfeb?auth=b9d4a42f4273e72d843f5fb8ffdf55de#159">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518168">test_community_undo_delete_message</a>, id: 702869 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div>Device 1: Find `Button` by `xpath`: `//*[@text="Undo"]` </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:112: in test_community_undo_delete_message
    self.channel.element_by_text("Undo").click()
../views/base_element.py:90: in click
    element = self.find_element()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: Button by xpath: `//*[@text="Undo"]` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6aa7dfe85eed432fba3c9756e2905ae6?auth=6e2f6e14061978ae423dad2396f8507c#106">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518180">test_community_mute_community_and_channel</a>, id: 703382 </b></td></tr><tr><td><p><blockquote><div>Device 1: Could not reach home view by pressing system back button </div><div>Device 1: Find `CommunitiesTab` by `accessibility id`: `communities-stack-tab` </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:125: in test_community_mute_community_and_channel
    self.home.communities_tab.click()
../views/base_element.py:90: in click
    element = self.find_element()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: CommunitiesTab by accessibility id: `communities-stack-tab` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6aa7dfe85eed432fba3c9756e2905ae6?auth=6e2f6e14061978ae423dad2396f8507c#122">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUiTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518183">test_1_1_chat_mute_chat</a>, id: 703496 </b></td></tr><tr><td><p><blockquote><div>Device 1: Looking for chat: 'receiver' </div><div>Device 1: Find `PreveiewMessageText` by `xpath`: `//*[@content-desc='author-primary-name'][starts-with(@text,'receiver')]/..//*[@content-desc='chat-message-text']` </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:628: in test_1_1_chat_mute_chat
    if not chat.chat_preview.text.startswith(muted_message):
../views/base_element.py:416: in text
    text = self.find_element().text
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: PreveiewMessageText by xpath: `//*[@content-desc='author-primary-name'][starts-with(@text,'receiver')]/..//*[@content-desc='chat-message-text']` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a3e6ccd4454949b4a8f2ce6042793449?auth=34954af1ca35ad67eb11353dc1ef4a49#103">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cb193260df294d61bc8f7de4750aaf59?auth=e0215e72d432ff3f9b60a9551aba003d#113">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518150">test_1_1_chat_delete_via_long_press_relogin</a>, id: 702784 </b></td></tr><tr><td><p><blockquote><div>Device 2: Looking for a message by text: test message </div><div>Device 2: Looking for a message by text: hey </div></blockquote></p>

```
None; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a3e6ccd4454949b4a8f2ce6042793449?auth=34954af1ca35ad67eb11353dc1ef4a49#145">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cb193260df294d61bc8f7de4750aaf59?auth=e0215e72d432ff3f9b60a9551aba003d#121">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518165">test_activity_center_contact_request_accept_swipe_mark_all_as_read</a>, id: 702851 </b></td></tr><tr><td><p><blockquote><div># STEP: Check that can accept contact request from read notifications</div><div>Device 1: Find `Button` by `accessibility id`: `selector-filter` </div></blockquote></p>

```
activity_center/test_activity_center.py:133: in test_activity_center_contact_request_accept_swipe_mark_all_as_read
    self.home_1.activity_unread_filter_button.click()
../views/base_element.py:90: in click
    element = self.find_element()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: Button by accessibility id: `selector-filter` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception; also Unexpected Alert is shown: 'Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0d3d0ff201d42f8972c1466aba62cdf?auth=785616eeca75eded9379690078eff5b7#99">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/36075f1b2077463fb4353c94b0484587?auth=51a4dd50c9af4ae5dbf8d2bf03780b17#110">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Expected to fail tests (1)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518184">test_community_discovery</a>, id: 703503 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Curated communities not loading, https://github.com/status-im/status-mobile/issues/17852]]</b><br/><br/></td></tr></tbody></table></details><h3>Passed tests (30)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518190">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/75a8fa5796b244f982160c94e4b031a9?auth=4835650a3a4276eaf3110e65d80635c1#39">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518191">test_wallet_add_remove_watch_only_account</a>, id: 727232 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/75a8fa5796b244f982160c94e4b031a9?auth=4835650a3a4276eaf3110e65d80635c1#103">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518171">test_activity_center_reply_read_unread_delete_filter_swipe</a>, id: 702947 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0dc6c169e2f44c9db56116706c8601ed?auth=00a16f6a5d3eb7fc078b37363e116a73#168">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cdac0c4f63b545529a66e99db5c8f52e?auth=52ad5454182adcdbfffa5e8293d53da3#123">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518170">test_navigation_jump_to</a>, id: 702936 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0dc6c169e2f44c9db56116706c8601ed?auth=00a16f6a5d3eb7fc078b37363e116a73#140">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cdac0c4f63b545529a66e99db5c8f52e?auth=52ad5454182adcdbfffa5e8293d53da3#120">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePRTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518174">test_activity_center_admin_notification_accept_swipe</a>, id: 702958 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cb5be9b2e82e4792892e423b6058a8c1?auth=43c912cc96d5b929761f55f81efe73b1#168">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b2d3a33534854187a026501cdbb48d30?auth=bd640cf2201220c803bf57188a58771c#117">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518173">test_activity_center_mentions</a>, id: 702957 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cb5be9b2e82e4792892e423b6058a8c1?auth=43c912cc96d5b929761f55f81efe73b1#120">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b2d3a33534854187a026501cdbb48d30?auth=bd640cf2201220c803bf57188a58771c#101">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518162">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8d9592540a604c658bdc6976b4880bc9?auth=c4c5a69ece11fb2d4678f7bbae34bbb5#323">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/40e7e7f757774041bab453be8980acbf?auth=1f7bfc32d803e4771748c670d1cee7cd#307">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518151">test_community_mentions_push_notification</a>, id: 702786 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8d9592540a604c658bdc6976b4880bc9?auth=c4c5a69ece11fb2d4678f7bbae34bbb5#133">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/40e7e7f757774041bab453be8980acbf?auth=1f7bfc32d803e4771748c670d1cee7cd#113">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518154">test_community_markdown_support</a>, id: 702809 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8d9592540a604c658bdc6976b4880bc9?auth=c4c5a69ece11fb2d4678f7bbae34bbb5#189">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/40e7e7f757774041bab453be8980acbf?auth=1f7bfc32d803e4771748c670d1cee7cd#178">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518172">test_community_hashtag_links_to_community_channels</a>, id: 702948 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8d9592540a604c658bdc6976b4880bc9?auth=c4c5a69ece11fb2d4678f7bbae34bbb5#326">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/40e7e7f757774041bab453be8980acbf?auth=1f7bfc32d803e4771748c670d1cee7cd#338">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518185">test_community_join_when_node_owner_offline</a>, id: 703629 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8d9592540a604c658bdc6976b4880bc9?auth=c4c5a69ece11fb2d4678f7bbae34bbb5#539">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/40e7e7f757774041bab453be8980acbf?auth=1f7bfc32d803e4771748c670d1cee7cd#525">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518166">test_1_1_chat_edit_message</a>, id: 702855 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cd6dfd37d3a549b3af0bd20e50d215b7?auth=9ee5b081b5ab9ff62bef8044eed85ab4#360">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cbf47048761948e1ad55ef35923db450?auth=ee26ba6a98cbdc7f92f7adbb05f407e4#264">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518140">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cd6dfd37d3a549b3af0bd20e50d215b7?auth=9ee5b081b5ab9ff62bef8044eed85ab4#88">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cbf47048761948e1ad55ef35923db450?auth=ee26ba6a98cbdc7f92f7adbb05f407e4#85">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518145">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cd6dfd37d3a549b3af0bd20e50d215b7?auth=9ee5b081b5ab9ff62bef8044eed85ab4#243">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cbf47048761948e1ad55ef35923db450?auth=ee26ba6a98cbdc7f92f7adbb05f407e4#175">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518155">test_1_1_chat_push_emoji</a>, id: 702813 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cd6dfd37d3a549b3af0bd20e50d215b7?auth=9ee5b081b5ab9ff62bef8044eed85ab4#305">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cbf47048761948e1ad55ef35923db450?auth=ee26ba6a98cbdc7f92f7adbb05f407e4#229">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518148">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 702782 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cd6dfd37d3a549b3af0bd20e50d215b7?auth=9ee5b081b5ab9ff62bef8044eed85ab4#140">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cbf47048761948e1ad55ef35923db450?auth=ee26ba6a98cbdc7f92f7adbb05f407e4#120">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518178">test_group_chat_reactions</a>, id: 703202 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/217ed82bf5ec4501b656d811de58b2ce?auth=f84a7c422bb33fd6cc3d96a050fcce1c#160">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/26a03bcb395648359c255998c411ac29?auth=35e1484739041a09955f357f308feb19#107">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5b2f2e65c12641e09c8869165ca788e8?auth=cbb8ea813aab7b6fe165af7de4a18c52#87">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518152">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/217ed82bf5ec4501b656d811de58b2ce?auth=f84a7c422bb33fd6cc3d96a050fcce1c#141">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/26a03bcb395648359c255998c411ac29?auth=35e1484739041a09955f357f308feb19#79">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5b2f2e65c12641e09c8869165ca788e8?auth=cbb8ea813aab7b6fe165af7de4a18c52#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518153">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/217ed82bf5ec4501b656d811de58b2ce?auth=f84a7c422bb33fd6cc3d96a050fcce1c#309">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/26a03bcb395648359c255998c411ac29?auth=35e1484739041a09955f357f308feb19#214">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5b2f2e65c12641e09c8869165ca788e8?auth=cbb8ea813aab7b6fe165af7de4a18c52#205">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518142">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/217ed82bf5ec4501b656d811de58b2ce?auth=f84a7c422bb33fd6cc3d96a050fcce1c#346">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/26a03bcb395648359c255998c411ac29?auth=35e1484739041a09955f357f308feb19#240">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5b2f2e65c12641e09c8869165ca788e8?auth=cbb8ea813aab7b6fe165af7de4a18c52#239">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518156">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6bc6d12842314a5d8ffa3e5a030de2df?auth=a702dff2b5678f835e018c1c99fdc9e7#154">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6a0931ee38bc4f2da6681990a55ddfeb?auth=b9d4a42f4273e72d843f5fb8ffdf55de#107">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518160">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6bc6d12842314a5d8ffa3e5a030de2df?auth=a702dff2b5678f835e018c1c99fdc9e7#172">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6a0931ee38bc4f2da6681990a55ddfeb?auth=b9d4a42f4273e72d843f5fb8ffdf55de#124">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518144">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6aa7dfe85eed432fba3c9756e2905ae6?auth=6e2f6e14061978ae423dad2396f8507c#77">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518163">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6aa7dfe85eed432fba3c9756e2905ae6?auth=6e2f6e14061978ae423dad2396f8507c#53">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518176">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6aa7dfe85eed432fba3c9756e2905ae6?auth=6e2f6e14061978ae423dad2396f8507c#143">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUiTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518149">test_1_1_chat_is_shown_message_sent_delivered_from_offline</a>, id: 702783 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a3e6ccd4454949b4a8f2ce6042793449?auth=34954af1ca35ad67eb11353dc1ef4a49#88">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cb193260df294d61bc8f7de4750aaf59?auth=e0215e72d432ff3f9b60a9551aba003d#85">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518189">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/66eee91219e94e25ad77d630684ffffc?auth=d1400603ed7b367952a05364b06c6f20#113">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/311c380af96f4542a8426c3fbd7def2a?auth=d715c553a733e81ec35cbb1b76ab15ec#92">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518188">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/66eee91219e94e25ad77d630684ffffc?auth=d1400603ed7b367952a05364b06c6f20#40">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/311c380af96f4542a8426c3fbd7def2a?auth=d715c553a733e81ec35cbb1b76ab15ec#40">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518164">test_activity_center_contact_request_decline</a>, id: 702850 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0d3d0ff201d42f8972c1466aba62cdf?auth=785616eeca75eded9379690078eff5b7#57">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/36075f1b2077463fb4353c94b0484587?auth=51a4dd50c9af4ae5dbf8d2bf03780b17#56">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3518147">test_add_contact_field_validation</a>, id: 702777 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0d3d0ff201d42f8972c1466aba62cdf?auth=785616eeca75eded9379690078eff5b7#120">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/36075f1b2077463fb4353c94b0484587?auth=51a4dd50c9af4ae5dbf8d2bf03780b17#184">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/quo/components/notifications/activity_log/view.cljs
+++ b/src/quo/components/notifications/activity_log/view.cljs
@@ -9,11 +9,10 @@
     [quo.components.tags.status-tags :as status-tags]
     [quo.foundations.colors :as colors]
     [react-native.core :as rn]
-    [reagent.core :as reagent]
     [utils.i18n :as i18n]))
 
 (defn- activity-reply-text-input
-  [{:keys [on-update-reply max-reply-length valid-reply?]} reply-input]
+  [{:keys [on-update-reply max-reply-length valid-reply?]} reply-input set-reply-input]
   [rn/view
    [rn/view
     {:style {:margin-top     16
@@ -26,12 +25,12 @@
      (i18n/label :t/your-answer)]
     [text/text
      {:style {:flex-shrink 1
-              :color       (if (valid-reply? @reply-input)
+              :color       (if (valid-reply? reply-input)
                              colors/neutral-40
                              colors/danger-60)}}
-     (str (count @reply-input) "/" max-reply-length)]]
+     (str (count reply-input) "/" max-reply-length)]]
    [input/input
-    {:on-change-text      #(do (reset! reply-input %)
+    {:on-change-text      #(do (set-reply-input %)
                                (when on-update-reply
                                  (on-update-reply %)))
      :auto-capitalize     :none
@@ -136,7 +135,7 @@
      (-> button
          (assoc :size size)
          (assoc :type subtype)
-         (assoc :disabled? (and replying? disable-when (disable-when @reply-input)))
+         (assoc :disabled? (and replying? disable-when (disable-when reply-input)))
          (assoc :inner-style
                 {:justify-content :center
                  :padding-bottom  0
@@ -153,17 +152,16 @@
     :blur?  blur?}])
 
 (defn- footer
-  [_]
-  (let [reply-input (reagent/atom "")]
-    (fn [{:keys [replying? items] :as props}]
-      [:<>
-       (when replying?
-         [activity-reply-text-input props reply-input])
-       (when items
-         [rn/view style/footer-container
-          (for [item items]
-            ^{:key (:key item)}
-            [footer-item-view item replying? reply-input])])])))
+  [{:keys [replying? items] :as props}]
+  (let [[reply-input set-reply-input] (rn/use-state "")]
+    [:<>
+     (when replying?
+       [activity-reply-text-input props reply-input set-reply-input])
+     (when items
+       [rn/view style/footer-container
+        (for [item items]
+          ^{:key (:key item)}
+          [footer-item-view item replying? reply-input])])]))
 
 (defn view
   [{:keys [icon

--- a/src/quo/components/notifications/count_down_circle.cljs
+++ b/src/quo/components/notifications/count_down_circle.cljs
@@ -48,7 +48,7 @@
   {:color {:dark  colors/neutral-80-opa-40
            :light colors/white-opa-40}})
 
-(defn circle-timer-internal
+(defn circle-timer
   [{:keys [color duration size stroke-width trail-color rotation initial-remaining-time]}]
   (let [theme                           (quo.theme/use-theme-value)
         rotation                        (or rotation :clockwise)
@@ -76,10 +76,7 @@
                                               (set-display-time
                                                (if completed? duration current-display-time))
                                               (when-not completed?
-                                                (reset! prev-frame-time frame-time)
-                                                (reset! frame-request
-                                                  (js/requestAnimationFrame
-                                                   swap-elapsed-time-each-frame))))))]
+                                                (reset! prev-frame-time frame-time)))))]
     (rn/use-effect #(reset! frame-request (js/requestAnimationFrame swap-elapsed-time-each-frame)))
     (rn/use-unmount #(js/cancelAnimationFrame @frame-request))
     [rn/view
@@ -101,8 +98,3 @@
           :stroke-width      stroke-width
           :stroke-dasharray  path-length
           :stroke-dashoffset (linear-ease display-time 0 path-length duration)}])]]))
-
-;;NOTE: for some reason it doesn't work without :f>
-(defn circle-timer
-  [props]
-  [:f> circle-timer-internal props])

--- a/src/quo/components/notifications/count_down_circle.cljs
+++ b/src/quo/components/notifications/count_down_circle.cljs
@@ -4,8 +4,7 @@
     [quo.foundations.colors :as colors]
     [quo.theme :as quo.theme]
     [react-native.core :as rn]
-    [react-native.svg :as svg]
-    [reagent.core :as reagent]))
+    [react-native.svg :as svg]))
 
 (defn- get-path-props
   [size stroke-width rotation]
@@ -49,60 +48,61 @@
   {:color {:dark  colors/neutral-80-opa-40
            :light colors/white-opa-40}})
 
-(defn- circle-timer-internal
-  [{:keys [color duration size stroke-width trail-color rotation initial-remaining-time theme]}]
-  (let [rotation                     (or rotation :clockwise)
-        duration                     (or duration 4)
-        stroke-width                 (or stroke-width 1)
-        size                         (or size 9)
-        max-stroke-width             (max stroke-width 0)
-        {:keys [path path-length]}   (get-path-props size max-stroke-width rotation)
-        start-at                     (get-start-at duration initial-remaining-time)
-        elapsed-time                 (reagent/atom 0)
-        prev-frame-time              (reagent/atom nil)
-        frame-request                (reagent/atom nil)
-        display-time                 (reagent/atom start-at)
-        ;; get elapsed frame time
-        swap-elapsed-time-each-frame (fn swap-elapsed-time-each-frame [frame-time]
-                                       (if (nil? @prev-frame-time)
-                                         (do (reset! prev-frame-time frame-time)
-                                             (reset! frame-request (js/requestAnimationFrame
-                                                                    swap-elapsed-time-each-frame)))
-                                         (let [delta                (- (/ frame-time 1000)
-                                                                       (/ @prev-frame-time 1000))
-                                               current-elapsed      (swap! elapsed-time + delta)
-                                               current-display-time (+ start-at current-elapsed)
-                                               completed?           (>= current-display-time duration)]
-                                           (reset! display-time (if completed?
-                                                                  duration
-                                                                  current-display-time))
-                                           (when-not completed?
-                                             (reset! prev-frame-time frame-time)
-                                             (reset! frame-request (js/requestAnimationFrame
-                                                                    swap-elapsed-time-each-frame))))))]
-    (reagent/create-class
-     {:component-will-unmount #(js/cancelAnimationFrame @frame-request)
-      :reagent-render
-      (fn []
-        (reset! frame-request (js/requestAnimationFrame swap-elapsed-time-each-frame))
-        [rn/view
-         {:style {:position :relative
-                  :width    size
-                  :height   size}}
-         [svg/svg
-          {:view-box (str "0 0 " size " " size)
-           :width    size
-           :height   size}
-          [svg/path
-           {:d path :fill :none :stroke (or trail-color :transparent) :stroke-width stroke-width}]
-          (when-not (= @display-time duration)
-            [svg/path
-             {:d                 path
-              :fill              :none
-              :stroke            (or color (get-in themes [:color theme]))
-              :stroke-linecap    :square
-              :stroke-width      stroke-width
-              :stroke-dasharray  path-length
-              :stroke-dashoffset (linear-ease @display-time 0 path-length duration)}])]])})))
+(defn circle-timer-internal
+  [{:keys [color duration size stroke-width trail-color rotation initial-remaining-time]}]
+  (let [theme                           (quo.theme/use-theme-value)
+        rotation                        (or rotation :clockwise)
+        duration                        (or duration 4)
+        stroke-width                    (or stroke-width 1)
+        size                            (or size 9)
+        max-stroke-width                (max stroke-width 0)
+        {:keys [path path-length]}      (get-path-props size max-stroke-width rotation)
+        start-at                        (get-start-at duration initial-remaining-time)
+        elapsed-time                    (rn/use-ref-atom 0)
+        prev-frame-time                 (rn/use-ref-atom nil)
+        frame-request                   (rn/use-ref-atom nil)
+        [display-time set-display-time] (rn/use-state start-at)
+        swap-elapsed-time-each-frame    (fn swap-elapsed-time-each-frame [frame-time]
+                                          (if (nil? @prev-frame-time)
+                                            (do (reset! prev-frame-time frame-time)
+                                                (reset! frame-request (js/requestAnimationFrame
+                                                                       swap-elapsed-time-each-frame)))
+                                            (let [delta                (- (/ frame-time 1000)
+                                                                          (/ @prev-frame-time 1000))
+                                                  current-elapsed      (swap! elapsed-time + delta)
+                                                  current-display-time (+ start-at current-elapsed)
+                                                  completed?           (>= current-display-time
+                                                                           duration)]
+                                              (set-display-time
+                                               (if completed? duration current-display-time))
+                                              (when-not completed?
+                                                (reset! prev-frame-time frame-time)
+                                                (reset! frame-request
+                                                  (js/requestAnimationFrame
+                                                   swap-elapsed-time-each-frame))))))]
+    (rn/use-effect #(reset! frame-request (js/requestAnimationFrame swap-elapsed-time-each-frame)))
+    (rn/use-unmount #(js/cancelAnimationFrame @frame-request))
+    [rn/view
+     {:style {:position :relative
+              :width    size
+              :height   size}}
+     [svg/svg
+      {:view-box (str "0 0 " size " " size)
+       :width    size
+       :height   size}
+      [svg/path
+       {:d path :fill :none :stroke (or trail-color :transparent) :stroke-width stroke-width}]
+      (when-not (= display-time duration)
+        [svg/path
+         {:d                 path
+          :fill              :none
+          :stroke            (or color (get-in themes [:color theme]))
+          :stroke-linecap    :square
+          :stroke-width      stroke-width
+          :stroke-dasharray  path-length
+          :stroke-dashoffset (linear-ease display-time 0 path-length duration)}])]]))
 
-(def circle-timer (quo.theme/with-theme circle-timer-internal))
+;;NOTE: for some reason it doesn't work without :f>
+(defn circle-timer
+  [props]
+  [:f> circle-timer-internal props])

--- a/src/quo/components/notifications/toast/view.cljs
+++ b/src/quo/components/notifications/toast/view.cljs
@@ -15,10 +15,10 @@
   [rn/touchable-highlight
    {:on-press       on-press
     :underlay-color :transparent}
-   [into
+   (into
     [rn/view
      {:style (merge (style/action-container theme) style)}]
-    children]])
+    children)])
 
 (defn toast-undo-action-internal
   [{:keys [undo-duration undo-on-press theme]}]

--- a/src/quo/components/selectors/filter/view.cljs
+++ b/src/quo/components/selectors/filter/view.cljs
@@ -3,23 +3,20 @@
     [quo.components.icon :as icon]
     [quo.components.selectors.filter.style :as style]
     [quo.theme :as quo.theme]
-    [react-native.core :as rn]
-    [reagent.core :as reagent]))
+    [react-native.core :as rn]))
 
-(defn view-internal
-  [initial-props]
-  (let [pressed? (reagent/atom (:pressed? initial-props))]
-    (fn [{:keys [blur? customization-color theme on-press-out]}]
-      [rn/touchable-without-feedback
-       {:accessibility-label :selector-filter
-        :on-press-out        (fn []
-                               (swap! pressed? not)
-                               (when on-press-out
-                                 (on-press-out @pressed?)))}
-       [rn/view {:style (style/container-outer customization-color @pressed? theme)}
-        [rn/view {:style (style/container-inner @pressed? blur? theme)}
-         [icon/icon :i/unread
-          {:color (style/icon-color @pressed? theme)
-           :size  20}]]]])))
-
-(def view (quo.theme/with-theme view-internal))
+(defn view
+  [{:keys [blur? customization-color on-press-out pressed?]}]
+  (let [theme                  (quo.theme/use-theme-value)
+        [pressed? set-pressed] (rn/use-state pressed?)
+        on-press-out           (fn []
+                                 (set-pressed (not pressed?))
+                                 (when on-press-out (on-press-out pressed?)))]
+    [rn/touchable-without-feedback
+     {:accessibility-label :selector-filter
+      :on-press-out        on-press-out}
+     [rn/view {:style (style/container-outer customization-color pressed? theme)}
+      [rn/view {:style (style/container-inner pressed? blur? theme)}
+       [icon/icon :i/unread
+        {:color (style/icon-color pressed? theme)
+         :size  20}]]]]))

--- a/src/quo/components/selectors/reactions_selector/view.cljs
+++ b/src/quo/components/selectors/reactions_selector/view.cljs
@@ -2,23 +2,20 @@
   (:require
     [quo.components.selectors.reaction-resource :as reactions.resource]
     [quo.components.selectors.reactions-selector.style :as style]
-    [react-native.core :as rn]
-    [reagent.core :as reagent]))
+    [react-native.core :as rn]))
 
 (defn view
-  [{:keys [start-pressed?]}]
-  (let [pressed? (reagent/atom start-pressed?)]
-    (fn [{:keys [emoji container-style on-press
-                 accessibility-label]
-          :or   {accessibility-label :reaction}}]
-      [rn/pressable
-       {:accessibility-label     accessibility-label
-        :allow-multiple-presses? true
-        :style                   (merge (style/container @pressed?)
-                                        container-style)
-        :on-press                (fn [e]
-                                   (swap! pressed? not)
-                                   (when on-press
-                                     (on-press e)))}
-       [rn/text
-        (reactions.resource/system-emojis emoji)]])))
+  [{:keys [emoji container-style on-press
+           accessibility-label start-pressed?]
+    :or   {accessibility-label :reaction}}]
+  (let [[pressed? set-pressed] (rn/use-state start-pressed?)
+        on-press               (fn [e]
+                                 (set-pressed (not pressed?))
+                                 (when on-press (on-press e)))]
+    [rn/pressable
+     {:accessibility-label     accessibility-label
+      :allow-multiple-presses? true
+      :style                   (merge (style/container pressed?)
+                                      container-style)
+      :on-press                on-press}
+     [rn/text (reactions.resource/system-emojis emoji)]]))

--- a/src/quo/components/selectors/selectors/view.cljs
+++ b/src/quo/components/selectors/selectors/view.cljs
@@ -3,47 +3,47 @@
     [quo.components.icon :as icons]
     [quo.components.selectors.selectors.style :as style]
     [quo.theme :as quo.theme]
-    [react-native.core :as rn]
-    [reagent.core :as reagent]))
-
-(defn- handle-press
-  [on-change checked-atom checked?]
-  (when checked-atom (swap! checked-atom not))
-  (when on-change (on-change (not checked?))))
+    [react-native.core :as rn]))
 
 (defn- base-selector
-  [{:keys [default-checked? checked?]}]
-  (let [controlled-component? (some? checked?)
-        internal-checked?     (when-not controlled-component?
-                                (reagent/atom (or default-checked? false)))]
-    (fn [{:keys [checked? disabled? blur? customization-color on-change container-style
-                 label-prefix outer-style-fn inner-style-fn icon-style-fn theme]
-          :or   {customization-color :blue}}]
-      (let [actual-checked?     (if controlled-component? checked? @internal-checked?)
-            accessibility-label (str label-prefix "-" (if actual-checked? "on" "off"))
-            test-id             (str label-prefix "-component")
-            outer-styles        (outer-style-fn {:checked?            actual-checked?
+  [{:keys [default-checked? checked? disabled? blur? customization-color on-change container-style
+           label-prefix outer-style-fn inner-style-fn icon-style-fn theme]
+    :or   {customization-color :blue}}]
+  (let [controlled-component?   (some? checked?)
+        [internal-checked?
+         set-internal-checked?] (rn/use-state (when-not controlled-component?
+                                                (or default-checked? false)))
+        actual-checked?         (if controlled-component? checked? internal-checked?)
+        accessibility-label     (str label-prefix "-" (if actual-checked? "on" "off"))
+        test-id                 (str label-prefix "-component")
+        outer-styles            (outer-style-fn {:checked?            actual-checked?
                                                  :disabled?           disabled?
                                                  :blur?               blur?
                                                  :container-style     container-style
                                                  :customization-color customization-color
-                                                 :theme               theme})]
-        [rn/pressable
-         (when-not disabled?
-           {:on-press                #(handle-press on-change internal-checked? actual-checked?)
-            :allow-multiple-presses? true})
-         [rn/view
-          {:style                             outer-styles
-           :needs-offscreen-alpha-compositing true
-           :accessibility-label               accessibility-label
-           :testID                            test-id}
-          [rn/view
-           {:style (inner-style-fn {:theme               theme
-                                    :checked?            actual-checked?
-                                    :blur?               blur?
-                                    :customization-color customization-color})}
-           (when (and icon-style-fn actual-checked?)
-             [icons/icon :i/check-small (icon-style-fn actual-checked? blur? theme)])]]]))))
+                                                 :theme               theme})
+        on-press                (rn/use-callback
+                                 (fn []
+                                   (when-not (nil? internal-checked?)
+                                     (set-internal-checked? (not internal-checked?)))
+                                   (when on-change (on-change (not actual-checked?))))
+                                 [internal-checked? actual-checked? on-change])]
+    [rn/pressable
+     (when-not disabled?
+       {:on-press                on-press
+        :allow-multiple-presses? true})
+     [rn/view
+      {:style                             outer-styles
+       :needs-offscreen-alpha-compositing true
+       :accessibility-label               accessibility-label
+       :testID                            test-id}
+      [rn/view
+       {:style (inner-style-fn {:theme               theme
+                                :checked?            actual-checked?
+                                :blur?               blur?
+                                :customization-color customization-color})}
+       (when (and icon-style-fn actual-checked?)
+         [icons/icon :i/check-small (icon-style-fn actual-checked? blur? theme)])]]]))
 
 (defn- toggle
   [props]

--- a/src/quo/components/tags/collectible_tag/schema.cljs
+++ b/src/quo/components/tags/collectible_tag/schema.cljs
@@ -8,7 +8,6 @@
       [:options {:optional true} [:maybe [:enum :add :hold]]]
       [:size {:optional true} [:maybe [:enum :size-24 :size-32]]]
       [:blur? {:optional true} [:maybe :boolean]]
-      [:theme :schema.common/theme]
       [:collectible-img-src :schema.common/image-source]
       [:collectible-name :string]
       [:collectible-id :string]]]]

--- a/src/quo/components/tags/collectible_tag/view.cljs
+++ b/src/quo/components/tags/collectible_tag/view.cljs
@@ -8,52 +8,49 @@
     [quo.theme]
     [react-native.core :as rn]
     [react-native.hole-view :as hole-view]
-    [reagent.core :as reagent]
     [schema.core :as schema]))
 
 (defn- view-internal
-  []
-  (let [container-width (reagent/atom 0)
-        on-layout       #(->> (oops/oget % :nativeEvent :layout :width)
-                              (reset! container-width))]
-    (fn [{:keys [options blur? theme collectible-img-src collectible-name collectible-id] :as props}]
-      (let [size (or (:size props) :size-24)]
-        [rn/view
-         {:on-layout on-layout}
-         [hole-view/hole-view
-          {:holes (if options
-                    [{:x            (- @container-width
-                                       (case size
-                                         :size-24 10
-                                         :size-32 12
-                                         nil))
-                      :y            (case size
-                                      :size-24 -6
-                                      :size-32 -4
-                                      nil)
-                      :width        16
-                      :height       16
-                      :borderRadius 8}]
-                    [])}
-          [rn/view {:style (style/container size options blur? theme)}
-           [rn/image {:style (style/collectible-img size) :source collectible-img-src}]
-           [text/text
-            {:size   :paragraph-2
-             :weight :medium
-             :style  (style/label theme)}
-            collectible-name]
-           [text/text
-            {:size        :paragraph-2
-             :weight      :medium
-             :margin-left 5
-             :style       (style/label theme)}
-            collectible-id]]]
-         (when options
-           [rn/view {:style (style/options-icon size)}
-            [icons/icon (if (= options :hold) :i/hold :i/add-token)
-             {:size     20
-              :no-color true}]])]))))
+  [{:keys [options blur? collectible-img-src collectible-name collectible-id] :as props}]
+  (let [theme                 (quo.theme/use-theme-value)
+        [container-width
+         set-container-width] (rn/use-state 0)
+        on-layout             (rn/use-callback
+                               #(set-container-width (oops/oget % :nativeEvent :layout :width)))
+        size                  (or (:size props) :size-24)]
+    [rn/view {:on-layout on-layout}
+     [hole-view/hole-view
+      {:holes (if options
+                [{:x            (- container-width
+                                   (case size
+                                     :size-24 10
+                                     :size-32 12
+                                     nil))
+                  :y            (case size
+                                  :size-24 -6
+                                  :size-32 -4
+                                  nil)
+                  :width        16
+                  :height       16
+                  :borderRadius 8}]
+                [])}
+      [rn/view {:style (style/container size options blur? theme)}
+       [rn/image {:style (style/collectible-img size) :source collectible-img-src}]
+       [text/text
+        {:size   :paragraph-2
+         :weight :medium
+         :style  (style/label theme)}
+        collectible-name]
+       [text/text
+        {:size        :paragraph-2
+         :weight      :medium
+         :margin-left 5
+         :style       (style/label theme)}
+        collectible-id]]]
+     (when options
+       [rn/view {:style (style/options-icon size)}
+        [icons/icon (if (= options :hold) :i/hold :i/add-token)
+         {:size     20
+          :no-color true}]])]))
 
-(def view
-  (quo.theme/with-theme
-   (schema/instrument #'view-internal component-schema/?schema)))
+(def view (schema/instrument #'view-internal component-schema/?schema))

--- a/src/quo/components/tags/tags.cljs
+++ b/src/quo/components/tags/tags.cljs
@@ -50,128 +50,131 @@
    - `fade-end-percentage` Percentage where fading starts relative to the total
   layout width of the `flat-list` data."
 
-  [{:keys [default-active fade-end-percentage]
-    :or   {fade-end-percentage 0.8}}]
-  (let [active-tab-id (reagent/atom default-active)
-        fading        (reagent/atom {:fade-end-percentage fade-end-percentage})
-        flat-list-ref (atom nil)]
-    (fn
-      [{:keys [data
-               fade-end-percentage
-               fade-end?
-               on-change
-               on-scroll
-               scroll-event-throttle
-               scrollable?
-               scroll-on-press?
-               size
-               type
-               labelled?
-               disabled?
-               blurred?
-               icon-color]
-        :or   {fade-end-percentage   fade-end-percentage
-               fade-end?             false
-               scroll-event-throttle 64
-               scrollable?           false
-               scroll-on-press?      false
-               size                  default-tab-size}
-        :as   props}]
-      (let [maybe-mask-wrapper (if fade-end?
+  [{:keys [default-active data fade-end-percentage fade-end? on-change on-scroll scroll-event-throttle
+           scrollable?
+           scroll-on-press? size type labelled? disabled? blurred? icon-color]
+    :or   {fade-end-percentage   0.8
+           fade-end?             false
+           scroll-event-throttle 64
+           scrollable?           false
+           scroll-on-press?      false
+           size                  default-tab-size}
+    :as   props}]
+  (let [[active-tab-id
+         set-active-tab-id] (rn/use-state default-active)
+        [fading set-fading] (rn/use-state {:fade-end-percentage fade-end-percentage})
+        flat-list-ref       (rn/use-ref-atom nil)
+        maybe-mask-wrapper  (rn/use-memo
+                             (fn []
+                               (if fade-end?
                                  [masked-view/masked-view
                                   {:mask-element (reagent/as-element
                                                   [linear-gradient/linear-gradient
                                                    {:colors         [:black :transparent]
-                                                    :locations      [(get @fading :fade-end-percentage)
+                                                    :locations      [(get fading :fade-end-percentage)
                                                                      1]
                                                     :start          {:x 0 :y 0}
                                                     :end            {:x 1 :y 0}
                                                     :pointer-events :none
                                                     :style          {:width  "100%"
                                                                      :height "100%"}}])}]
-                                 [:<>])]
-        (if scrollable?
-          (conj
-           maybe-mask-wrapper
-           [rn/flat-list
-            (merge
-             (dissoc props
-              :default-active
-              :fade-end-percentage
-              :fade-end?
-              :on-change
-              :scroll-on-press?
-              :size)
-             (when scroll-on-press?
-               {:initial-scroll-index (utils.collection/first-index #(= @active-tab-id (:id %)) data)})
-             {:ref #(reset! flat-list-ref %)
-              :extra-data (str @active-tab-id)
-              :horizontal true
-              :scroll-event-throttle scroll-event-throttle
-              :shows-horizontal-scroll-indicator false
-              :data data
-              :key-fn (comp str :id)
-              :on-scroll (fn [^js e]
-                           (when fade-end?
-                             (let [offset-x       (oget e "nativeEvent.contentOffset.x")
-                                   content-width  (oget e "nativeEvent.contentSize.width")
-                                   layout-width   (oget e "nativeEvent.layoutMeasurement.width")
-                                   new-percentage (calculate-fade-end-percentage
-                                                   {:offset-x            offset-x
-                                                    :content-width       content-width
-                                                    :layout-width        layout-width
-                                                    :max-fade-percentage fade-end-percentage})]
-                               ;; Avoid unnecessary re-rendering.
-                               (when (not= new-percentage (get @fading :fade-end-percentage))
-                                 (swap! fading assoc :fade-end-percentage new-percentage))))
-                           (when on-scroll
-                             (on-scroll e)))
-              :render-fn (fn [{:keys [id label resource]} index]
-                           [rn/view
-                            {:style {:margin-right  (if (= size default-tab-size) 12 8)
-                                     :padding-right (when (= index (dec (count data)))
-                                                      (get-in props [:style :padding-left]))}}
-                            [tag/tag
-                             {:id         id
-                              :size       size
-                              :active     (= id @active-tab-id)
-                              :resource   resource
-                              :blurred?   blurred?
-                              :icon-color icon-color
-                              :disabled?  disabled?
-                              :label      (if labelled?
-                                            label
-                                            (when (= type :label) label))
-                              :type       type
-                              :labelled?  labelled?
-                              :on-press   (fn [id]
-                                            (reset! active-tab-id id)
-                                            (when scroll-on-press?
-                                              (.scrollToIndex ^js @flat-list-ref
-                                                              #js
-                                                               {:animated     true
-                                                                :index        index
-                                                                :viewPosition 0.5}))
-                                            (when on-change
-                                              (on-change id)))}]])})])
-          [rn/view {:style {:flex-direction :row}}
-           (for [{:keys [id label resource]} data]
-             ^{:key id}
-             [rn/view {:style {:margin-right 8}}
-              [tag/tag
-               (merge {:id         id
-                       :size       size
-                       :type       type
-                       :label      (if labelled?
-                                     label
-                                     (when (= type :label) label))
-                       :active     (= id active-tab-id)
-                       :disabled?  disabled?
-                       :blurred?   blurred?
-                       :icon-color icon-color
-                       :labelled?  (if (= type :label) true labelled?)
-                       :resource   (if (= type :icon)
-                                     :i/placeholder
-                                     resource)
-                       :on-press   #(do (reset! active-tab-id %)
-                                        (when on-change (on-change %)))})]])])))))
+                                 [:<>]))
+                             [fade-end? fading])
+        on-scroll           (rn/use-context
+                             (fn [^js e]
+                               (when fade-end?
+                                 (let [offset-x       (oget e "nativeEvent.contentOffset.x")
+                                       content-width  (oget e "nativeEvent.contentSize.width")
+                                       layout-width   (oget e "nativeEvent.layoutMeasurement.width")
+                                       new-percentage (calculate-fade-end-percentage
+                                                       {:offset-x            offset-x
+                                                        :content-width       content-width
+                                                        :layout-width        layout-width
+                                                        :max-fade-percentage fade-end-percentage})]
+                                   ;; Avoid unnecessary re-rendering.
+                                   (when (not= new-percentage (get fading :fade-end-percentage))
+                                     (set-fading (assoc fading :fade-end-percentage new-percentage)))))
+                               (when on-scroll
+                                 (on-scroll e)))
+                             [fade-end? fading fade-end-percentage])
+        data-count          (count data)
+        style-padding-left  (get-in props [:style :padding-left])
+        render-fn           (rn/use-callback
+                             (fn [{:keys [id label resource]} index]
+                               [rn/view
+                                {:style {:margin-right  (if (= size default-tab-size) 12 8)
+                                         :padding-right (when (= index (dec data-count))
+                                                          style-padding-left)}}
+
+                                [tag/tag
+                                 {:id         id
+                                  :size       size
+                                  :active     (= id active-tab-id)
+                                  :resource   resource
+                                  :blurred?   blurred?
+                                  :icon-color icon-color
+                                  :disabled?  disabled?
+                                  :label      (if labelled?
+                                                label
+                                                (when (= type :label) label))
+                                  :type       type
+                                  :labelled?  labelled?
+                                  :on-press   (fn [id]
+                                                (set-active-tab-id id)
+                                                (when scroll-on-press?
+                                                  (.scrollToIndex ^js @flat-list-ref
+                                                                  #js
+                                                                   {:animated     true
+                                                                    :index        index
+                                                                    :viewPosition 0.5}))
+                                                (when on-change
+                                                  (on-change id)))}]])
+                             [size default-tab-size data-count style-padding-left active-tab-id blurred?
+                              icon-color disabled? labelled? type scroll-on-press? on-change])
+        on-press            (rn/use-callback #(do (set-active-tab-id %) (when on-change (on-change %)))
+                                             [on-change])
+        key-fn              (rn/use-callback (comp str :id))
+        ref                 (rn/use-callback #(reset! flat-list-ref %))
+        clean-props         (dissoc props
+                             :default-active
+                             :fade-end-percentage
+                             :fade-end?
+                             :on-change
+                             :scroll-on-press?
+                             :size)]
+    (if scrollable?
+      (conj
+       maybe-mask-wrapper
+       [rn/flat-list
+        (merge
+         clean-props
+         (when scroll-on-press?
+           {:initial-scroll-index (utils.collection/first-index #(= active-tab-id (:id %)) data)})
+         {:ref                               ref
+          :extra-data                        (str active-tab-id)
+          :horizontal                        true
+          :scroll-event-throttle             scroll-event-throttle
+          :shows-horizontal-scroll-indicator false
+          :data                              data
+          :key-fn                            key-fn
+          :on-scroll                         on-scroll
+          :render-fn                         render-fn})])
+
+      [rn/view {:style {:flex-direction :row}}
+       (for [{:keys [id label resource]} data]
+         ^{:key id}
+         [rn/view {:style {:margin-right 8}}
+          [tag/tag
+           {:id         id
+            :size       size
+            :type       type
+            :label      (if labelled?
+                          label
+                          (when (= type :label) label))
+            :active     (= id active-tab-id)
+            :disabled?  disabled?
+            :blurred?   blurred?
+            :icon-color icon-color
+            :labelled?  (if (= type :label) true labelled?)
+            :resource   (if (= type :icon) :i/placeholder resource)
+            :on-press   on-press}]])])))

--- a/src/quo/components/tags/token_tag/view.cljs
+++ b/src/quo/components/tags/token_tag/view.cljs
@@ -7,10 +7,9 @@
     [quo.components.utilities.token.view :as token]
     [quo.theme :as quo.theme]
     [react-native.core :as rn]
-    [react-native.hole-view :as hole-view]
-    [reagent.core :as reagent]))
+    [react-native.hole-view :as hole-view]))
 
-(defn- view-internal
+(defn view
   "Options:
     - :options - false / :add / :hold (default false)
     - :size - :size-24 / :size-32 (default :size-24)
@@ -18,45 +17,44 @@
     - :theme - :light / :dark
     - :token-value - string - token value
     - :token-symbol - string"
-  []
-  (let [container-width (reagent/atom 0)]
-    (fn [{:keys [options size blur? theme token-value token-img-src token-symbol]
-          :or   {size :size-24}}]
-      [rn/view
-       {:on-layout #(reset! container-width
-                      (oget % :nativeEvent :layout :width))}
-       [hole-view/hole-view
-        {:holes (if options
-                  [{:x            (- @container-width
-                                     (case size
-                                       :size-24 10
-                                       :size-32 12
-                                       nil))
-                    :y            (case size
-                                    :size-24 -6
-                                    :size-32 -4
-                                    nil)
-                    :width        16
-                    :height       16
-                    :borderRadius 8}]
-                  [])}
-        [rn/view {:style (style/container size options blur? theme)}
-         [token/view
-          {:style        (style/token-img size)
-           :token        token-symbol
-           :size         (case size
-                           :size-24 :size-20
-                           :size-32 :size-28)
-           :image-source token-img-src}]
-         [text/text
-          {:size   :paragraph-2
-           :weight :medium
-           :style  (style/label theme)}
-          (str token-value " " token-symbol)]]]
-       (when options
-         [rn/view {:style (style/options-icon size)}
-          [icons/icon (if (= options :hold) :i/hold :i/add-token)
-           {:size     20
-            :no-color true}]])])))
-
-(def view (quo.theme/with-theme view-internal))
+  [{:keys [options size blur? token-value token-img-src token-symbol]
+    :or   {size :size-24}}]
+  (let [theme                 (quo.theme/use-theme-value)
+        [container-width
+         set-container-width] (rn/use-state 0)
+        on-layout             (rn/use-callback #(set-container-width
+                                                 (oget % :nativeEvent :layout :width)))]
+    [rn/view {:on-layout on-layout}
+     [hole-view/hole-view
+      {:holes (if options
+                [{:x            (- container-width
+                                   (case size
+                                     :size-24 10
+                                     :size-32 12
+                                     nil))
+                  :y            (case size
+                                  :size-24 -6
+                                  :size-32 -4
+                                  nil)
+                  :width        16
+                  :height       16
+                  :borderRadius 8}]
+                [])}
+      [rn/view {:style (style/container size options blur? theme)}
+       [token/view
+        {:style        (style/token-img size)
+         :token        token-symbol
+         :size         (case size
+                         :size-24 :size-20
+                         :size-32 :size-28)
+         :image-source token-img-src}]
+       [text/text
+        {:size   :paragraph-2
+         :weight :medium
+         :style  (style/label theme)}
+        (str token-value " " token-symbol)]]]
+     (when options
+       [rn/view {:style (style/options-icon size)}
+        [icons/icon (if (= options :hold) :i/hold :i/add-token)
+         {:size     20
+          :no-color true}]])]))

--- a/src/status_im/common/toasts/view.cljs
+++ b/src/status_im/common/toasts/view.cljs
@@ -61,4 +61,4 @@
   (->> (rf/sub [:toasts])
        :ordered
        (into [rn/view {:style (style/outmost-transparent-container)}]
-             (map #(with-meta [:f> f-container %] {:key %})))))
+             (map #(with-meta [f-container %] {:key %})))))

--- a/src/status_im/navigation/view.cljs
+++ b/src/status_im/navigation/view.cljs
@@ -54,7 +54,6 @@
 
 (defn screen
   [screen-key]
-
   (reagent.core/reactify-component
    (fn []
      (let [screen-details              (get (if js/goog.DEBUG
@@ -110,7 +109,7 @@
             sheet])]]))
    functional-compiler))
 
-(def toasts (reagent/reactify-component toasts/toasts))
+(def toasts (reagent/reactify-component toasts/toasts functional-compiler))
 
 (def alert-banner
   (reagent/reactify-component

